src/v/storage/kvstore.h

```diff
@@ -111,6 +111,8 @@ class kvstore {
     ss::future<> start();
     ss::future<> stop();
 
+    ss::future<> collect_startup_garbage();
```

Comment:
nit: Is a better name something like: `do_startup_compaction()`?

nit: This needs some documentation, maybe add some of the comments about why this doesn't happen during `start` here as well?

Reply:
> `do_startup_compaction()`

Wasn't sure about the correct nomenclature here, like are we doing compaction or just collecting junk files? Anyway, this is certainly easier on the eyes.

> documentation

Done

Reply:
> Wasn't sure about the correct nomenclature here

I'll defer to anyone on the storage team to what would be considered "correct", but it seems like compaction to me. Naming is hard tho 🤷